File: Python/CHAPITRE7/exo7-2.py
n = int(input("Entrer la taille du tableau : "))
tab = []
for i in range(n):
    val = int(input(f"tab[{i}] = "))
    tab.append(val)
for i in range(n):
    print(f"tab[{i}] = {tab[i]}")

indice = int(input("Entrer l'indice du tableau a afficher : "))
answer = False

while not answer:
    if 0 <= indice < n:
        print(f"tab[{indice}] = {tab[indice]}")
        answer = True
        break
    else:
        indice=int(input(f"Indice invalide. Veuillez entrer un indice entre 0 et {n-1} : "))
        answer = False

File: C/CHAPITRE7/exo7-2.c
#include <stdio.h>
#include <stdlib.h>

int main() 
{
    int n, i;

    // Saisie de la taille
    printf("Entrez la taille du tableau : ");
    scanf("%d", &n);

    // Allocation dynamique
    int *tab = (int *)malloc(n * sizeof(int));

    if (tab == NULL) {
        printf("Erreur d'allocation memoire.\n");
        return 1;
    }

    // Saisie des elements
    printf("Entrez les %d elements du tableau :\n", n);
    for (int j = 0; j < n; j++) 
    {
        printf("  tab[%d] = ", j);
        scanf("%d", tab + j);  // utilisation des pointeurs
    }

    // Saisie de l'indice avec verification
    do 
    {
        printf("Entrez un indice i : ");
        scanf("%d", &i);

        if (i < 0 || i >= n) {
            printf("Indice invalide ! Reessayez.\n");
        }

    } while (i < 0 || i >= n);

    // Affichage
    printf("tab[%d] = %d\n", i, *(tab + i));

    // Liberation memoire
    free(tab);

    return 0;
}